datalad/core/distributed/push.py

```diff
@@ -523,20 +533,17 @@ def _push(dspath, content, target, data, force, jobs, res_kwargs, pbars,
                 'branch'
             )
             return
-        if is_annex_repo:
-            # we could face a managed branch, in which case we need to
-            # determine the actual one and make sure it is sync'ed with the
-            # managed one, and push that one instead. following methods can
-            # be called unconditionally
-            repo.localsync(managed_only=True)
-            active_branch = repo.get_corresponding_branch(
-                active_branch) or active_branch
-        refspecs2push.append(
-            # same dance as above
-            active_branch
-            if ds.config.get('branch.{}.merge'.format(active_branch), None)
-            else '{ab}:{ab}'.format(ab=active_branch)
-        )
+
+    # make sure that we always push the active branch (the context for the
+    # potential path arguments) and the annex branch -- because we claim
+    # to know better than any git config
+    must_have_branches = [active_branch] if active_branch else []
+    if is_annex_repo:
+        must_have_branches.append('git-annex')
+    for branch in must_have_branches:
+        # active_branch could be None
```

Comment:
Not true with the current upstream code :)

Reply:
Arrgh, true. Too many iterations if moving conditionals around. Will fix that up.